test/functional/sqlsrv/sqlsrv_900_output_param_memory_data.phpt

```diff
@@ -0,0 +1,81 @@
+--TEST--
+GitHub issue 900 - output parameter displays data from memory when not finalized
+--DESCRIPTION--
+This test verifies that when there is an active resultset and output parameter not finalized, it should not show any data from client memory. This test does not work with AlwaysEncrypted because
+the output param is not assigned in the stored procedure.
+--ENV--
+PHPT_EXEC=true
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$size = 30;
+
+function getOutputParam($conn, $storedProcName, $inout, $isVarchar, $isMax)
+{
+    global $size;
+    
+    $output = null;
+    $dir = ($inout)? SQLSRV_PARAM_INOUT : SQLSRV_PARAM_OUT;
+    $dataType = ($isVarchar)? "SQLSRV_SQLTYPE_VARCHAR" : "SQLSRV_SQLTYPE_NVARCHAR";
+    $sqlType = ($isMax)? call_user_func($dataType, 'max') : call_user_func($dataType, $size);
+
+    $stmt = sqlsrv_prepare($conn, "$storedProcName @OUTPUT = ?", array(array(&$output, $dir, null, $sqlType)));
+    if (!$stmt) {
+        fatalError("getOutputParam: failed when preparing to call $storedProcName");
+    }
+    if (!sqlsrv_execute($stmt)) {
+        fatalError("getOutputParam: failed to execute procedure $storedProcName");
+    }
+
+    // The output param should be doubled in size for wide characters;
+    // for max fields it should be the maximum anticipated size 8000.
+    // However, it should not contain any data so after trimming it
+    // should be merely an empty string
+    $len = strlen($output);
+    $expectedLen = ($isMax)? 8000 :  ($size * 2);

```

Comment:
Why is there no comparison to the max size of 8000 in the pdo test?

Reply:
good point! I should redesign the sqlsrv test to fix the size like the pdo test